linux/devdoc/threadpool_linux_requirements.md

```diff
@@ -0,0 +1,350 @@
+
+MOCKABLE_FUNCTION(, int, threadpool_timer_restart, TIMER_INSTANCE_HANDLE, timer, uint32_t, start_delay_ms, uint32_t, timer_period_ms);
+
+MOCKABLE_FUNCTION(, void, threadpool_timer_cancel, TIMER_INSTANCE_HANDLE, timer);
+
+MOCKABLE_FUNCTION(, void, threadpool_timer_destroy, TIMER_INSTANCE_HANDLE, timer);
+
+```
+###  threadpool_create
+
+```C
+MOCKABLE_FUNCTION(, THREADPOOL_HANDLE, threadpool_create, EXECUTION_ENGINE_HANDLE, execution_engine);
+```
+
+`threadpool_create` creates a new threadpool.
+
+`threadpool_create` shall allocate memory for a threadpool object and on success return a non-`NULL` handle to it. 
+
+If `execution_engine` is `NULL`, `threadpool_create` shall fail and return `NULL`. 
+
+`threadpool_create` shall create a `SM_HANDLE` by calling `sm_create`.
+
+`threadpool_create` shall initialize thread count parameters by calling `execution_engine_linux_get_parameters` with parameter `execution_engine`. 
+
+`threadpool_create` shall allocate memory for an array of thread objects and on success return a non-`NULL` handle to it.
+
+`threadpool_create` shall allocate memory for an array of tasks and on success return a non-`NULL` handle to it.
+
+`threadpool_create` shall initialize every task item in the tasks array with `task_func` and `task_param` set to `NULL` and `task_state` set to `TASK_NOT_USED`.
+
+`threadpool_create` shall create a SRW lock by calling `srw_lock_create`.
+
+`threadpool_create` shall create a shared semaphore between threads with initialized value zero.
+
+`insert_idx`  and `consume_idx` for the task array shall be intialzied to 0.
+
+If any error occurs, `threadpool_create` shall fail and return `NULL`.
+
+### threadpool_destroy
+
+```C
+MOCKABLE_FUNCTION(, void, threadpool_destroy, THREADPOOL_HANDLE, threadpool);
+```
+
+`threadpool_destroy` frees the resouces associated with `threadpool`.
+
+If `threadpool` is `NULL`, `threadpool_destroy` shall return.
+
+`threadpool_destroy` shall implicit close if `threadpool` is open. 
+
+`threadpool_destroy` shall free the resources associated with the threadpool handle. 
+
+`threadpool_destroy` shall destroy the semphore by calling `sem_destroy`.
+
+`threadpool_destroy` shall destroy the SRW lock by calling `srw_lock_destroy`.
+
+### threadpool_open_async
+
+```C
+MOCKABLE_FUNCTION(, int, threadpool_open_async, THREADPOOL_HANDLE, threadpool, ON_THREADPOOL_OPEN_COMPLETE, on_open_complete, void*, on_open_complete_context);
+```
+
+`threadpool_open_async` opens the threadpool asynchronously.
+
+If `threadpool` is `NULL`, `threadpool_open_async` shall fail and return a non-zero value. 
+
+`threadpool_open_async` shall call `sm_open_begin`. 
+
+`threadpool_open_async` shall create the threads for `threadpool` using `ThreadAPI_Create`.
+
+If `threadpool` has already been opened, `threadpool_open_async` shall fail and return a non-zero value.
```

Comment:
Missing "If any error occurs, threadpool_open_async shall fail and return a non-zero value" We need that to cover the case that ThreadAPI_Create fails.

Reply:
added